salsa_dancing_molecules/scatter_plot.py

```diff
@@ -12,8 +12,48 @@ def draw_scatter_plot(x_axis, y_axis, xlabel, ylabel, filename):
     File is saved as filename (which should include .png).
     """
     # Agg -> use non-interactive backend to save plot to file
-    matplotlib.use('Agg')
-    plt.scatter(x_axis, y_axis)
+    # matplotlib.use('Agg')
+    plt.scatter(x_axis, y_axis, s=1)
     plt.xlabel(xlabel)
     plt.ylabel(ylabel)
     plt.savefig(filename)
+    plt.show()
```

Comment:
show() only works on systems that can show windows. I think it should be optional. This comment applies to the draw_four_scatter_plots() function as well.